doc/build/dts/bindings-syntax.rst

```diff
@@ -438,6 +439,47 @@ can write this property as follows:
        type: phandle-array
        specifier-space: mbox
 
+pattern
+=======
+
+A property with ``pattern: true`` indicates that the property name is a regular
+expression. This can be used to match an arbitrary number of properties
+following the same naming convention. For example:
+
+.. code-block:: YAML
+
+   # base.yaml
+
+   properties:
+     ".*-supply$":
+        type: phandle
+        pattern: true
+
+This will match any property name ending in ``-supply``. For example:
+
+.. code-block:: devicetree
+
+   my-node {
+           foo-supply = <&foo_regulator>;
+           bar-supply = <&bar_regulator>;
+   };
+
+Properties with a name matching another pattern property can be used to set
+constraints or override certain options. For example:
+
+.. code-block:: YAML
+
+   include: base.yaml
+
+   properties:
+     # matches '.*-supply$' from base.yaml, will be required.
+     vin-supply:
+       required: true
+
+This is allowed for the ``required``, ``const``, ``enum`` and
+``specifier-space`` options. Other options including ``type`` and ``deprecated``
+will be taken from the pattern property.
```

Comment:
nit: This could be updated, since `type` is a special case.

Reply:
reworded this a bit, in the end, they are not overridable even if they can be re-specified again.